packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4229,18 +4241,627 @@ class InputDecoration {
 
 /// Defines the default appearance of [InputDecorator]s.
 ///
-/// This class is used to define the value of [ThemeData.inputDecorationTheme].
-/// The [InputDecorator], [TextField], and [TextFormField] widgets use
-/// the current input decoration theme to initialize null [InputDecoration]
-/// properties.
+/// Descendant widgets obtain the current theme's [InputDecorationThemeData] using
+/// [InputDecorationTheme.of]. When a widget uses [InputDecorationTheme.of], it is
+/// automatically rebuilt if the theme later changes.
+///
+/// See also:
+///
+///  * [InputDecorationThemeData], which describes the actual configuration of an
+///    input decoration theme.
+///  * [ThemeData.inputDecorationTheme], which specifies an input decoration theme as
+///    part of the overall Material theme.
+class InputDecorationTheme extends InheritedTheme with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.cardTheme].
```

Comment:
```suggestion
  /// Creates a theme that can be used for [ThemeData.inputDecorationTheme].
```

Reply:
Oh this might not be accurate. It should be used for InputDecorationThemeData class instead of here.